internal/node/machinehash_test.go

```diff
@@ -0,0 +1,158 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package node
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/cartesi/rollups-node/internal/deps"
+	"github.com/cartesi/rollups-node/internal/machine"
+	"github.com/cartesi/rollups-node/pkg/addresses"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/suite"
+)
+
+const BlockchainHttpEndpoint = "http://0.0.0.0:" + deps.DefaultDevnetPort
+
+type ValidateMachineHashSuite struct {
+	suite.Suite
+}
+
+func TestValidateMachineHash(t *testing.T) {
+	suite.Run(t, new(ValidateMachineHashSuite))
+}
+
+func (s *ValidateMachineHashSuite) TestItFailsWhenSnapshotHasNoHash() {
+	machineDir, err := os.MkdirTemp("", "")
+	defer os.RemoveAll(machineDir)
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+
+	err = validateMachineHash(context.Background(), machineDir, "", "")
+	s.ErrorContains(err, "no such file or directory")
+}
+
+func (s *ValidateMachineHashSuite) TestItFailsWhenHashHasWrongSize() {
+	machineDir, err := mockMachineDir("deadbeef")
+	defer os.RemoveAll(machineDir)
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+
+	err = validateMachineHash(context.Background(), machineDir, "", "")
+	s.ErrorContains(err, "wrong size")
+}
+
+func (s *ValidateMachineHashSuite) TestItFailsWhenContextIsCanceled() {
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	defer cancel()
+	machineDir, err := createMachineSnapshot()
+	defer os.RemoveAll(machineDir)
+	if err != nil {
+		s.FailNow(err.Error())
+	}
+	devnet, err := startDevnet()
+	if err != nil {
+		s.Fail(err.Error())
+		if err = deps.Terminate(context.Background(), devnet); err != nil {
+			s.Fail(err.Error())
+		}
+	}
```

Comment:
can you try something like that?
```suggestion
	s.Require().Nil(err)
	defer func() {
		err := deps.Terminate(context.Background(), devnet)
		s.Require().Nil(err)
	}()
```

Reply:
I just realized it made no sense in the first place since the failing function was the one starting the containers, so there's no container to stop after all.

Reply:
The defer recommendation still stands.